src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md

```diff
@@ -520,6 +520,18 @@ The author of this interface has deprecated implementing this interface.
 |CodeFix|False|
 ---
 
+## RS1043: Do not use file types for analyzers, generators, and code fixers
+
+Using a 'file' type is not allowed for analyzers, generators, or code fixers.
```

Comment:
Can we add a bit more as to _why_ this is here? As is this just states what it does. No hint here as to why we added this enforcement for people who see this in the future.